src/compas_fab/ghpython/components/Cf_VisualizeRobot/code.py

```diff
@@ -34,13 +35,21 @@ def RunScript(self, robot, group, configuration, attached_collision_meshes, show
                 collision = robot.artist.draw_collision()
 
             if show_base_frame:
-                base_frame = frames[0]
+                base_compas_frame = compas_frames[0]
+                artist = FrameArtist(base_compas_frame)
+                base_frame = artist.draw()
 
             if show_end_effector_frame:
-                ee_frame = robot.forward_kinematics(configuration, group, options=dict(solver='model'))
-
-            if not show_frames:
-                frames = None
+                ee_compas_frame = robot.forward_kinematics(configuration, group, options=dict(solver='model'))
+                artist = FrameArtist(ee_compas_frame)
+                ee_frame = artist.draw()
+
+            if show_frames:
+                frames = []
```

Comment:
isn't `frames` undefined at line 62 if `show_frames` is `False`?

Reply:
It's initialized to `None` in line 20.

Reply:
aha! sorry, didn't see it 👍